### PR TITLE
Prepare to implement default filters on the api

### PR DESCRIPTION
**Pull request by mbenadda**
## Purpose

As we were working on #292, we noticed some necessary improvements we should undertake before getting to the heart of the matter.

## Proposal

- [x] simplify indexers by using `@staticmethod` & `@classmethod` instead of instantiating them
- [x] make `ES_INDEXES` a named tuple instead of an array: we're going to need the names, it ensures validity for downstream devs, and it remains easy to consume
- [x] import indexers in viewsets through settings
- [x] break in our new viewset/list design with the simple orgs & subjects viewsets

### CODE REVIEW

File: src/richie/apps/search/viewsets/courses.py [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
 
 from elasticsearch.exceptions import NotFoundError
 from rest_framework.response import Response
 from rest_framework.viewsets import ViewSet
 
 from ..forms import CourseListForm
-from ..indexers.courses import CoursesIndexer
+
+# Get the courses indexer from the settings
+indexer = import_string(settings.ES_INDICES.courses)
```
Comment (jmaupetit) [inline src/richie/apps/search/viewsets/courses.py]: I think naming it `CoursesIndexer` was more explicit; why did you change it to a generic name?

Comment (mbenadda) [reply src/richie/apps/search/viewsets/courses.py]: I can name it `courses_indexer` but not `CoursesIndexer` :)

Comment (jmaupetit) [reply src/richie/apps/search/viewsets/courses.py]: :sweat: True. Let's do this.

File: src/richie/apps/search/indexers/subjects.py [hunk 2]
```diff
@@ -55,3 +57,40 @@ def format_es_subject_for_api(es_subject, best_language):
...
+        # Build a query that matches on the name field if it was handed by the client
+        # Note: test_elasticsearch_feature.py needs to be updated whenever the search call
+        # is updated and makes use new features.
+        if params_form.cleaned_data.get("query"):
+            query = {
+                "query": {
+                    "match": {
+                        "name.fr": {
+                            "query": params_form.cleaned_data.get("query"),
+                            "analyzer": "french",
```
Comment (sampaccoud) [inline src/richie/apps/search/indexers/subjects.py]: it looks like this should be configurable but maybe not in the frame of this PR.

File: tests/apps/search/test_indexers_subjects.py [hunk 3]
```diff
@@ -115,3 +121,42 @@ def test_format_es_subject_for_api(self):
             SubjectsIndexer.format_es_subject_for_api(es_subject, "en"),
             {"id": 89, "image": "example.com/image.png", "name": "Computer science"},
         )
+
+    def test_build_es_query(self):
```
Comment (sampaccoud) [inline tests/apps/search/test_indexers_subjects.py]: this was not tested yet or you forgot to remove the existing test code?

Comment (mbenadda) [reply tests/apps/search/test_indexers_subjects.py]: The return values for the viewsets were tested (the viewsets were previously building the queries).

Now it's tested more granularly on the indexer

Comment (sampaccoud) [reply tests/apps/search/test_indexers_subjects.py]: so you may want to change the test on the viewset to test what is now relevant to test? I'm thinking for example mocking the call to build_es_query and raising the exception to check that the response has the error?

Comment (sampaccoud) [reply tests/apps/search/test_indexers_subjects.py]: and stop testing the format of the query there... ?

Comment (mbenadda) [reply tests/apps/search/test_indexers_subjects.py]: Right now the tests on the viewset methods are perhaps "less obvious" (more integration than unit) but also definitely less coupled to the internal implementation than they would be with a mock. Those cases you mention are covered by the existing tests.

I do not think it is worth it to change these tests to move testing upstream and increase coupling when we have existing tests covering what we want to cover.

Also note it would be simple enough for subjects & orgs but the courses test is 600 line long. In my opinion it would not be a good investment to rewrite them on the indexers just for the sake of it.


I suggest we keep them as they are and move them to the indexers only if/when the necessity arises

Comment (sampaccoud) [reply tests/apps/search/test_indexers_subjects.py]: I follow you.

File: tests/apps/search/test_viewsets_organizations.py [hunk 4]
```diff
@@ -225,4 +161,5 @@ def test_search_organizations_with_invalid_params(self):
 
         # The client received a BadRequest response with the relevant data
         self.assertEqual(response.status_code, 400)
+        print(response.data["errors"])
```
Comment (sampaccoud) [inline tests/apps/search/test_viewsets_organizations.py]: print fail
